File: LeetCode/1007/Solution.cpp
class Solution {
public:
    int minDominoRotations(vector<int>& tops, vector<int>& bottoms) {
        int ret = 100000;
        
        int base = tops[0];
        int cnt = 0;
        for(int j=1;j<tops.size();++j) {
            if(tops[j] == base) {
                continue;
            } else if(bottoms[j] == base) {
                cnt++;
            } else {
                cnt = 100000;
                break;
            }
        }
        ret = min(ret, cnt);
            
        cnt = 1;
        for(int j=1;j<tops.size();++j) {
            if(bottoms[j] == base) {
                continue;
            } else if(tops[j] == base) {
                cnt++;
            } else {
                cnt = 100000;
                break;
            }
        }
        ret = min(ret, cnt);
        
        base = bottoms[0];
        cnt = 0;
        for(int j=1;j<tops.size();++j) {
            if(bottoms[j] == base) {
                continue;
            } else if(tops[j] == base) {
                cnt++;
            } else {
                cnt = 100000;
                break;
            }
        }
        ret = min(ret, cnt);
            
        cnt = 1;
        for(int j=1;j<tops.size();++j) {
            if(tops[j] == base) {
                continue;
            } else if(bottoms[j] == base) {
                cnt++;
            } else {
                cnt = 100000;
                break;
            }
        }
        ret = min(ret, cnt);

        return ret == 100000 ? -1 : ret;
    }
};

File: LeetCode/1007/Solution.java
class Solution {
    public int minDominoRotations(int[] tops, int[] bottoms) {
        int n = tops.length;
        int result = Integer.MAX_VALUE;
        for(int i=1;i<=6;++i) {
            int topSwaps = 0;
            int bottomSwaps = 0;
            boolean valid = true;
            for(int j=0;j<n;++j) {
                if(tops[j] != i && bottoms[j] != i) {
                    valid = false;
                    break;
                }
                if(tops[j] != i) {
                    topSwaps++;
                }
                if(bottoms[j] != i) {
                    bottomSwaps++;
                }
            }
            if(valid) {
                result = Math.min(result, Math.min(topSwaps, bottomSwaps));
            }
        }
        return result == Integer.MAX_VALUE ? -1 : result;
    }
}
